djl-easy/src/main/java/ai/djl/easy/cv/ImageClassification.java

```diff
@@ -81,12 +90,59 @@ private ImageClassification() {}
         return ModelZoo.loadModel(criteria.build());
     }
 
-    /*
-    I am leaving this commented out as an example of what the DJL-Easy train API should look like.
-    public static <I> ZooModel<I, Classifications> train(Class<I> input, Dataset dataset, Performance performance) {
-        throw new UnsupportedOperationException("Not yet implemented");
-    }
+    /**
+     * Trains the recommended image classification model on a custom dataset.
+     *
+     * @param datasetFactory to build the datasets to train on and validate against
+     * @param performance to determine the desired model tradeoffs
+     * @return the model as a {@link ZooModel} with the {@link Translator} included
+     * @throws IOException if the dataset could not be loaded
+     * @throws TranslateException if the translator has errors
      */
+    public static ZooModel<Image, Classifications> train(
+            ImageClassificationDatasetFactory datasetFactory, Performance performance)
+            throws IOException, TranslateException {
+
+        int channels = datasetFactory.getImageChannels();
+        int width = datasetFactory.getImageWidth();
+        int height = datasetFactory.getImageHeight();
+        Shape imageShape = new Shape(channels, height, width);
+        List<String> classes = datasetFactory.getClasses();
+
+        // Determine the layers based on performance
+        int numLayers = performance.switchPerformance(18, 50, 152);
+
+        Block block =
+                ResNetV1.builder()
+                        .setImageShape(imageShape)
+                        .setNumLayers(numLayers)
+                        .setOutSize(classes.size())
+                        .build();
+        Model model = Model.newInstance("ImageClassification");
+        model.setBlock(block);
+
+        TrainingConfig trainingConfig =
+                new DefaultTrainingConfig(Loss.softmaxCrossEntropyLoss())
+                        .addEvaluator(new Accuracy())
+                        .addTrainingListeners(TrainingListener.Defaults.basic());
+
+        try (Trainer trainer = model.newTrainer(trainingConfig)) {
+            trainer.initialize(new Shape(1).addAll(imageShape));
+            EasyTrain.fit(
+                    trainer,
+                    35,
```

Comment:
Why hardcode 35?

Reply:
This probably could get more attention, but I googled it and there was a paper that determined that 35 epochs was sufficient. Later on, it may be worth replacing this with a more specific HPO system anyway